### ISSUE TITLE
Customization of "Envoy Filter" to make an external call inside a Lua code 

### ISSUE THREAD

**Issue by vraghani** [istio/istio#53940]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

We have implemented a “custom side car injection” (in Istio-addon deployed in AKS Cluster) by using EnvoyFilter. 
As per Istio documentation ([Istio / Envoy Filter](https://istio.io/latest/docs/reference/config/networking/envoy-filter/)) , we have implemented example 2 , to intercept all HTTP calls and inject with Lua code to call an external service.

We extended custom LUA script to call an external service . As per documentation, for calling ext services requires a cluster definition in envoy. We implemented this and get an error
While deploying this yaml file we get an error as below:

**Admission webhook has denied the request for envoy filter to create a cluster definition** 
Can you pls help on this issue ?

### Version

```Text
Kubernetes version 1.29
Istio add-on 1.23
```

**Comment by howardjohn** [comment]
I don't work on AKS but AFAIK they do not allow EnvoyFilter in their addon

**Comment by keithmattix** [comment]
The addon does allow Lua now. Can you add your service as an extensionProvider and then reference that cluster in Lua? https://istio.io/latest/docs/tasks/security/authorization/authz-custom/#define-the-external-authorizer

**Comment by kyessenov** [comment]
I think it's actually better to define a service entry instead. There is a canonical encoding of cluster names for all service entries, that you can reference from Lua.

**Comment by keithmattix** [comment]
Fair point - +1 to service entry

**Comment by kyessenov** [comment]
> > I think it's actually better to define a service entry instead. There is a canonical encoding of cluster names for all service entries, that you can reference from Lua.
> 
> Hey @kyessenov are you referring to this canonical name? I'm subscribed to the thread as well. Thanks in advance
> 

No, I meant the xDS name for the cluster. It usually looks like `outbound|<port>|<host>|...`.  You can see it in the config dump. These names are effectively stable, but not properly documents in the user facing docs.
